fix(stock): make "at risk of expiration" flag work

jniles opened the pull request:
This fix restores the functionality of the "at risk of expiration" flag
on the articles in stock registry.  It does not affect the API, just the
internal calculations.

In addition, we also changed the behavior of the articles in stock
action dropdown menu so the link to the stock lots now filters on the
depot and time period in question.

![image](https://user-images.githubusercontent.com/896472/109151831-f7a5a980-776a-11eb-8044-0097b0aacd81.png)


Closes #5422.

jniles committed:
fix(stock): make "at risk of expiration" flag work

This fix restores the functionality of the "at risk of expiration" flag
on the articles in stock registry.  It does not affect the API, just the
internal calculations.

In addition, we also changed the behavior of the articles in stock
action dropdown menu so the link to the stock lots now filters on the
depot and time period in question.

Closes #5422.

jmcameron reviewed (commented):
When I tested this with bhima_test and selected the "Show Only Inventories with Risky Lots", I noticed that the filter tag did not get translated:
![image](https://user-images.githubusercontent.com/62145/109184500-ea28f900-7743-11eb-9003-593dda2c9ed3.png)

Then I tested with the IMCK db (downloaded 2021-01-08), did yarn migrate, and CALL zRecomputeStockMovementStatus().

Again, it found no records and the filter tag is not translated.  
ALSO when I go back to the search filters, the flag for "Show only inventories with risky lots" already clear (not being cached).

BUT if I clear the filter and search for Inventory Item 'Vitamin B6..." (code 100118), it DOES show as at risk of expiration.
![image](https://user-images.githubusercontent.com/62145/109185216-a4b8fb80-7744-11eb-8435-d8e70e85f676.png)

So there are still some problems here.

jmcameron reviewed (approved):
Based on the updates, my second round of tests, and a quick review of the code, it looks good to me.

jniles committed:
chore: fix CMM calculation tests

This commit fixes the CMM calculation tests now that we have gone
through a month of 28 days.
